sql/parse/parse.go

```diff
@@ -439,12 +439,30 @@ func convertExplain(ctx *sql.Context, n *sqlparser.Explain) (sql.Node, error) {
 }
 
 func convertPrepare(ctx *sql.Context, n *sqlparser.Prepare) (sql.Node, error) {
-	childStmt, err := sqlparser.Parse(n.Expr)
+	expr := n.Expr
+	if strings.HasPrefix(n.Expr, "@") {
```

Comment:
seems like the answer is no per our convo

Reply:
I think I see what you mean, but the way it's currently implemented the user var types would never show up in the AST as the "inner" query would just be the variable.

`prepare stmt from @a` would make an inner query that's just `@a`.

Given how MySQL throws errors with the value of the user variable rather than the user variable itself, I suspect they do something similar and replace the variables with the actual values and then parse.